core/src/main/kotlin/DokkaBootstrapImpl.kt

```diff
@@ -11,8 +12,16 @@ import java.util.function.BiConsumer
 class DokkaBootstrapImpl : DokkaBootstrap {
 
     class DokkaProxyLogger(val consumer: BiConsumer<String, String>) : DokkaLogger {
-        override var warningsCount: Int = 0
-        override var errorsCount: Int = 0
+        private val warningsCounter = AtomicInteger()
+        private val errorsCounter = AtomicInteger()
+
+        override var warningsCount: Int
+            get() = warningsCounter.get()
+            set(value) = warningsCounter.set(value)
+
+        override var errorsCount: Int
+            get() = errorsCounter.get()
+            set(value) = errorsCounter.set(value)
```

Comment:
I thought about extracting this common logic into a separate AtomicCounterDokkaLogger class that encapsulates the counter increments, but I'm not sure it's worth it (new API + possible compatibility problems).

Can do if you like it more, though

Reply:
Up to you. I have no preferences.